src/store/widgets/reducers.ts

```diff
@@ -62,3 +75,14 @@ function menuActionHandler(state, payload) {
     },
   };
 }
+
+// todo here: update only when the annotations have changed
```

Comment:
this would probably require splitting `annotationIds` update to a separate action type.

On the other hand, if the `annotationIds` passed here is the same(by ref) array as in previous state, it should be fine

Reply:
Oh, that was an unclear comment.
What I meant was that location should not be changed unless the viewer's annotations have changed -- a simple solution I used in the previous architecture that stopped the window from changing location every time the cursor slipped off the window for a split second

Reply:
You mean a scenario when a user hovers over the highlight, which results in the `Viewer` window getting moved by a couple of pixels every time?

So to avoid it - if annotation id is same as before, only toggle visibility (without modifying the { x, y } location) ?

Reply:
Precisely :) (i'll update the comment)